### PR TITLE
feat(react-dialog): removes `DialogSurface` native dialog support 

### PR DESCRIPTION
**Pull request by bsunderhus**
## Current Behavior

After many discussions, the implementation of native `<dialog>` as a valid element to be used by `DialogSurface` component presented as desirable.

## New Behavior

The PR https://github.com/microsoft/fluentui/pull/24525 had valid points presented by @layershifter  and @Hotell  about our matrix support and Safari issues with native `<dialog>` element. After some investigations, the support for `<dialog>` native element as default element for `DialogSurface` has proven to be a challenge (mostly due to Safari's lack of support for `HTMLDialogElement` modal methods and also `::backdrop` pseudoelement).

With that in mind, this PR reverts `DialogSurface` to only support `div`. Having in mind that perhaps in future iterations whenever we have a better browser support this might be added again.

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/Avatar-mount_base.html">1253</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/Avatar-mount.html">1283</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/Button-mount_base.html">947</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/Button-mount.html">939</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/FluentProvider-mount_base.html">1485</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/FluentProvider-mount.html">1481</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/FluentProviderWithTheme-mount_base.html">577</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/FluentProviderWithTheme-mount.html">574</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">533</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">542</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">578</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">585</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/MakeStyles-mount_base.html">1931</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/MakeStyles-mount.html">1948</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/SpinButton-mount_base.html">2353</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24979/perf-test-react-components/SpinButton-mount.html">2328</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `188.681 kB`<br />`52.366 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.394 kB`<br />`11.007 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |  `82.755 kB`<br />`24.581 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.851 kB`<br />`1.964 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/adea89e6cc51161fbb544f465e298f60046fd55c'>adea89e6cc51161fbb544f465e298f60046fd55c</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts [hunk 1]
```diff
@@ -13,8 +13,11 @@ import type {
   DialogSurfaceState,
 } from './DialogSurface.types';
 import { useDialogContext_unstable } from '../../contexts';
-import { isEscapeKeyDismiss, HTMLDialogElementProps } from '../../utils';
+import { isEscapeKeyDismiss } from '../../utils';
 import { useModalAttributes } from '@fluentui/react-tabster';
+
+const defaultDialogType: 'div' | 'dialog' = 'div';
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts:19]:
```suggestion
```

File: packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts [hunk 2]
```diff
@@ -118,27 +72,18 @@ export const useDialogSurface_unstable = (
   return {
     components: {
       backdrop: 'div',
-      root: 'dialog',
+      root: defaultDialogType,
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts:75]:
```suggestion
      root: 'div',
```

File: packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts [hunk 3]
```diff
@@ -118,27 +72,18 @@ export const useDialogSurface_unstable = (
...
-            role: modalType === 'alert' ? 'alertdialog' : undefined,
-            onClose: handleNativeClose,
-            onClick: handleNativeClick,
-            onCancel: handleNativeCancel,
-          }
-        : {
-            'aria-modal': modalType !== 'non-modal',
-            role: modalType === 'alert' ? 'alertdialog' : 'dialog',
-          }),
+    root: getNativeElementProps(as ?? defaultDialogType, {
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts:84]:
```suggestion
    root: getNativeElementProps(as ?? 'div', {
```

**Review by layershifter — APPROVED** [review]

**Pull request merged**